DEVELOPING_A_RUNTIME.md

```diff
@@ -1,10 +1,30 @@
 # Runtime Developer Reference
 
-The following page is a reference for how to create a Runtime using the available Runtime API.
+The following page is a reference for how to create a Runtime by implementing
+the Runtime API interface.
+
+A Runtime is an npm module that implements the following interface:
+
+```typescript
+interface Runtime {
+  version: number;
+  build: (options: BuildOptions) => Promise<BuildResult>;
+  analyze?: (options: AnalyzeOptions) => Promise<string>;
+  prepareCache?: (options: PrepareCacheOptions) => Promise<CacheOutputs>;
+  shouldServe?: (options: ShouldServeOptions) => Promise<boolean>;
+  startDevServer?: (
+    options: StartDevServerOptions
+  ) => Promise<StartDevServerResult>;
+}
+```
+
+The `version` property and the `build()` function are the only _required_ fields.
+The rest are optional extensions that a Runtime _may_ implement in order to
+enhance functionality. These functions are documented in more detail below.
+
+Official Runtimes are published to [the npm registry](https://npmjs.com) as a package and referenced in the `use` property of the `vercel.json` configuration file.
 
-A Runtime is an npm module that exposes a `build` function and optionally an `analyze` function and `prepareCache` function.
-Official Runtimes are published to [npmjs.com](https://npmjs.com) as a package and referenced in the `use` property of the `vercel.json` configuration file.
-However, the `use` property will work with any [npm install argument](https://docs.npmjs.com/cli/install) such as a git repo url which is useful for testing your Runtime.
+> **Note:** The `use` property in the `builds` array will work with any [npm install argument](https://docs.npmjs.com/cli/install) such as a git repo URL, which is useful for testing your Runtime.
```

Comment:
Should we also mention the `functions` property? https://vercel.com/docs/configuration#project/functions

Reply:
Ya for sure. Will add.